Add documentation for custom clustering and multiplex_filtering into webpages (#923)
xia2/xia2: doc/sphinx/multiplex/basic_usage.rst

+++++++++++++++++++++++++++++
Basic Usage of xia2.multiplex
+++++++++++++++++++++++++++++
-------------
Tutorial Data
-------------

> The data (~6GB total) were collected on I24 at Diamond Light Source and is available is available to download from |cows_pigs_people_data|.

Reviewer: ```suggestion
The data (~6GB total) were collected on I24 at Diamond Light Source and is available to download from |cows_pigs_people_data|.
```

This tutorial assumes knowledge of processing with DIALS. For more information, see this `DIALS tutorial`_.

::

    dials.import data/*gz
    dials.find_spots imported.expt
    dials.index imported.expt strong.refl joint=False
    dials.refine indexed.expt indexed.refl
    dials.integrate refined.expt refined.refl

--------------
xia2.multiplex
--------------
``xia2.multiplex`` can now be run on this data. First, the software will do some preliminary unit cell clustering to
remove any datasets with a very different unit cell. Next, consistent symmetry will be applied and any indexing ambiguities are
resolved. Note that this uses ``dials.cosym`` under the hood. Next, the data is scaled, and the space group is determined.
Once the data is merged, a range of analyses are performed which are useful for diagnostics. Let's explore this on the example data.

::

    xia2.multiplex integrated.expt integrated.refl

Note that if you have separate integrated files you can list them in the input command.

::

    xia2.multiplex integrated_1.expt integrated_1.refl integrated_2.expt integrated_2.refl

> There are also some additional commandline options you may find useful:

Reviewer: ```suggestion
There are also some additional command-line options you may find useful:
```

 * ``symmetry.space_group`` - set the space group if known, which can speed up symmetry analysis. 
 * ``resolution.d_min`` - set the resolution cut-off, otherwise it will be automatically determined.
 * ``filtering.method`` - applies additional filtering, see :doc:`here <multiplex_filtering>`.
 * ``clustering.output_clusters`` - output identified sub-clusters, see :doc:`here <intensity_based_clustering>` for more choices and details.

----------------------------------
Diagnostic Tools in xia2.multiplex
----------------------------------

> Most diagnostic graph are available in ``xia2.multiplex.html``. Some extra plots are output as ``.png`` files to the working folder.

Reviewer: ```suggestion
Most diagnostic graphs are available in ``xia2.multiplex.html``. Some extra plots are output as ``.png`` files to the working folder.
```

**Unit cell analysis:**
The unit cell of each crystal input to ``xia2.multiplex`` is plotted to give an initial analysis of isomorphism. To see the
unit cell distributions of all input data, open `cluster_unit_cell_p1`. If you see subgroups in the dendrogram denoted by
different colours, this means there are some major differences in unit cell and only the largest will be considered for analysis 
with xia2.multiplex. If the entire dendrogram is the same colour, then all datasets will be fed into the pipeline. The ``xia2.multiplex.html``
log file also contains graphs related to unit cell clustering. It is important to note that these graphs only pertain to any subset 
chosen by the initial unit cell filtering. Subtle differences in unit cell can still occur, and the initial unit cell filtering
is fairly conservative, so these graphs are worth examining. In this case, all the data were accepted during the initial filtering,
and are shown to have very similar unit cells.

.. image:: ../figures/multiplex_uc.png

**Orientation analysis:**

> Multi-crystal data collections often have small wedges either due to radiation damage contraints, or sample environment

Reviewer: ```suggestion
Multi-crystal data collections often have small wedges either due to radiation damage constraints, or sample environment
```

> constraints (ie *in situ* data collections). A risk with such experiments are that all the crystals are measured in the same

Reviewer: ```suggestion
constraints (i.e. *in situ* data collections). A risk with such experiments are that all the crystals are measured in the same
```

orientation, and the same small volumes of reciprocal space are measured repeatedly. Coverage of reciprocal space can be
assessed by considering the distributions of the unit cell axes in reciprocal space. A stereographic projection plot is used
for this. If the dots are widely distributed around the circles, then there is wide coverage of reciprocal space and
preferential orientation is not an issue.

.. image:: ../figures/multiplex_sp.png

:math:`{\Delta}CC1/2`:
Another useful tool is the :math:`{\Delta}CC1/2` analysis. This shows whether individual datasets contribute positively,
or negatively to the overall quality of the merged dataset. Note that any outliers are not removed unless ``filtering.method=deltacchalf``

> is used. In this case, the :math:`{\Delta}CC1/2` for each dataset is quite minimal, although one dataset has a stronger affect

Reviewer: ```suggestion
is used. In this case, the :math:`{\Delta}CC1/2` for each dataset is quite minimal, although one dataset has a stronger effect
```
